Created a new branch and added stw_read_yaml function

nikdata opened the pull request.

nikdata committed:
adding ignore files

nikdata committed:
merge upstream/master

nikdata committed:
merge upstream/master

nikdata committed:
initial commit of the stw_read_yaml user-facing function

ijlyttle commented on R/stw_read_yaml.R:
```diff
@@ -0,0 +1,40 @@
+#' Process YAML dictionary
```
Maybe something like "Create metadata object from YAML"

ijlyttle commented on R/stw_read_yaml.R:
```diff
@@ -0,0 +1,40 @@
+#' Process YAML dictionary
+#'
+#' @param file_name location of yaml file
```
perhaps this could be called `file`

ijlyttle commented on R/stw_read_yaml.R:
```diff
@@ -0,0 +1,40 @@
+#'
+#' @param file_name location of yaml file
+#'
+#' @return object with S3 class `stw_meta`
```
consider using something like
```r
#' @inherit return stw_meta
```

ijlyttle commented on R/stw_read_yaml.R:
```diff
@@ -0,0 +1,40 @@
+
+  # process the dictioanry
+  grande <- lapply(infile$dictionary, function(x) {
+    as.tibble(x)
```
please use `tibble::as_tibble()`

ijlyttle reviewed (changes requested):
please rename `nik-scratch` to `scratch` and adjust `.Rbuildignore` and `.gitignore`

nikdata committed:
resolved issues around syntax and package call-outs

nikdata committed:
fixing my errors on the Roxygen

nikdata committed:
fixed file name

nikdata committed:
fixed conflict in Description and updating from master

Merge remote-tracking branch 'refs/remotes/upstream/master'

Conflicts:
	DESCRIPTION

ijlyttle commented:
Thanks Nik!

ijlyttle commented:
Sorry Nik!
